app/map.js

```diff
@@ -0,0 +1,205 @@
+var app = require('remote').require('app');
+var jetpack = require('fs-jetpack').cwd(app.getAppPath());
+
+function buildShaderProgram(gl) {
+    var vertexShader = gl.createShader(gl.VERTEX_SHADER);
+    gl.shaderSource(vertexShader, jetpack.read("../app/vertex.glsl"));
+    gl.compileShader(vertexShader);
+    if (!gl.getShaderParameter(vertexShader, gl.COMPILE_STATUS)) {
+        throw new Error("Failed to compile vertex shader: " + gl.getShaderInfoLog(vertexShader));
+    }
+
+    var fragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+    gl.shaderSource(fragmentShader, jetpack.read("../app/fragment.glsl"));
+    gl.compileShader(fragmentShader);
+    if (!gl.getShaderParameter(fragmentShader, gl.COMPILE_STATUS)) {
+        throw new Error("Failed to compile fragment shader: " + gl.getShaderInfoLog(fragmentShader));
+    }
+
+    var program = gl.createProgram();
+    gl.attachShader(program, vertexShader);
+    gl.attachShader(program, fragmentShader);
+    gl.linkProgram(program);
+    if (!gl.getProgramParameter(program, gl.LINK_STATUS)) {
+        throw new Error("Failed to link shader program: " + gl.getProgramInfoLog(program));
+    }
+
+    return program;
+}
+
+function buildTileDataTexture(data) {

```

Comment:
It's building a texture where each pixel actually represents a tile index. The texture, though, is colors, defined as 4 x 8bit ints. If we want to support more than 256 possible tiles, we've got to break the actual tile index apart into its constituent bytes, and then store then as consecutive Uint8s in the array. 

(t >> 8) % 256 is just "get the byte starting at bit 8".